pkg/utils/controller/placement_resolver_test.go

```diff
@@ -101,39 +103,62 @@ func TestResolvePlacementFromKey(t *testing.T) {
 
 			placement, err := FetchPlacementFromKey(context.Background(), fakeClient, tt.placementKey)
 
-			if tt.expectError {
-				assert.Error(t, err)
-				assert.Nil(t, placement)
+			if tt.expectedErr != nil {
+				if err == nil {
+					t.Fatalf("Expected error but got nil")
+				}
+				if !errors.Is(err, tt.expectedErr) {
+					t.Fatalf("Expected error: %v, but got: %v", tt.expectedErr, err)
+				}
+				return
+			}
+
+			if tt.expectedErr != nil {
+				if placement != nil {
+					t.Fatalf("Expected nil placement but got: %v", placement)
+				}
 				return
 			}
 
-			require.NoError(t, err)
-			require.NotNil(t, placement)
+			if placement == nil {
+				t.Fatalf("Expected placement but got nil")
+			}
 
 			// Determine if this is a cluster-scoped placement based on namespace
 			isCluster := placement.GetNamespace() == ""
-			assert.Equal(t, tt.expectCluster, isCluster)
-			assert.Equal(t, tt.expectedName, placement.GetName())
-			assert.Equal(t, tt.expectedNS, placement.GetNamespace())
+			if isCluster != tt.expectCluster {
+				t.Errorf("Expected cluster-scoped: %v, got: %v", tt.expectCluster, isCluster)
+			}
+
+			if diff := cmp.Diff(tt.expectedName, placement.GetName()); diff != "" {
+				t.Errorf("Name mismatch (-want +got):\n%s", diff)
+			}
+			if diff := cmp.Diff(tt.expectedNS, placement.GetNamespace()); diff != "" {
+				t.Errorf("Namespace mismatch (-want +got):\n%s", diff)
+			}
```

Comment:
The validation can be simplified by comparing the whole structure. you can refer to the code in https://github.com/kubefleet-dev/kubefleet/pull/90#discussion_r2137179762 and we don't need to list all the want fields one by one.